datashape/promote.py

```diff
@@ -7,7 +7,7 @@
 __all__ = ['promote', 'optionify']
 
 
-def promote(lhs, rhs):
+def promote(lhs, rhs, promote_option=True):
```

Comment:
What do you think about not promoting option types at all and having client code that calls this use `optionify`?